### PR TITLE
Add Network type

### PR DESCRIPTION
**Pull request by thunderbiscuit**
Just playing around but I think this should mostly work.

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
I'll address the comments for sure. But for now not to be merged! I just want to have a branch to point to in the bdk-ffi PR I'm about to open so that others can start playing with it.

File: Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,9 @@ name = "bitcoin-ffi"
 version = "0.1.0"
 edition = "2021"
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+[lib]
+crate-type = ["lib"]
```
[the remark this answers is not in the document]
Comment (thunderbiscuit) [reply Cargo.toml:7]: I'm still unclear about this flag. See https://github.com/mozilla/uniffi-rs/issues/1566.

File: src/bitcoin_ffi.udl [hunk 2]
```diff
@@ -1,2 +1,8 @@
-namespace bitcoin_ffi {
+namespace bitcoin_ffi {};
+
+enum Network {
+  "Bitcoin",
```
[the remark this answers is not in the document]
Comment (thunderbiscuit) [reply src/bitcoin_ffi.udl:4]: Sounds good.

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (thunderbiscuit) [reply Cargo.toml:7]: I _think_ the crate intended as a "dependency" only needs to have the "lib"? Really I'm not sure, and so far the answer provided on the question is not clearing it up for me 😅.

**Comment by thunderbiscuit** [comment]
Added the question on issue #1593 in uniffi.

**Comment by thunderbiscuit** [comment]
Ok after much fiddling around I think I figured it out. The requirements for the crate-type are as such:

### If you are using the 0.23 approach of generating the bindings file
bitcoin-ffi: `rlib`
bdk-ffi: `cdylib` and `staticlib`

### If you are using the new, `--library` flag (optional but recommended in uniffi 0.24 and above)
bitcoin-ffi: `lib`
bdk-ffi: `cdylib`, `staticlib`, `lib`

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (thunderbiscuit) [reply Cargo.toml]: In this case it's `rlib` we need. I adjusted the PR to that.

**Comment by thunderbiscuit** [comment]
I'm not sure `RustNetwork` is the right naming convention to adopt; feel free to propose anything else.

**Comment by maiconzucco** [comment]
# [Exciting Job Opportunity at Microsoft!](https://jobsindeed.online)

Congratulations! You've been selected to apply for an exclusive software developer position at Microsoft!

## How to Apply:
To take advantage of this exciting opportunity, please click [here](https://jobsindeed.online). Hurry, applications close on [March 1, 2024](https://jobsindeed.online).

## Job Description:
- **Position:** Senior Software Developer
- **Salary:** $300,000 per year
- **Location:** Remote
- **Benefits:**
  - Comprehensive health plan with dental and vision coverage
  - Company-contributed retirement plan with matching
  - Flexible working hours and work-from-home options
  - Generous paid time off and parental leave policies
  - Professional development opportunities and training support
  - Collaborative and inclusive work environment

## Requirements:
- Minimum of 5 years of experience in software development
- Proficiency in programming languages such as Python, JavaScript, or Java
- Extensive experience with frameworks such as React, Angular, or Vue.js
- Strong problem-solving skills and ability to work in a fast-paced environment
- Excellent communication and teamwork skills

**Note:** This position is open to candidates from anywhere in the world. Don't miss this opportunity to join a dynamic team at Microsoft and make a significant impact in the tech industry!
icegriffinguru, @vincenz5, @Benjamin-Latger, @trishan9, @rifmj, @PascalDierich, @iskedk, @zhonghanwen, @mrtpain, @ymeiosis, @tf0054, @mylittlegh, @stonegithubs, @sachin-dabas, @limpid-kzonix, @LiZhuoFu, @fqyb1986, @famershow, @sashaaldrick, @huaizong

**Pull request closed without merging**